### PR TITLE
Rename FHIR Client

### LINKED ISSUE #1401: Rename FhirHttpClient and FhirClient
Rename FhirClient to LegacyFhirClient, including the  namespace
Rename FhirHttpClient to FhirClient, including the namespace

### PR DESCRIPTION
**Pull request by mmsmits**
fixes #1401

### CODE REVIEW

File: src/Hl7.Fhir.Specification.Tests/Source/ResourceResolverTests.cs [hunk 1]
```diff
@@ -88,7 +89,7 @@ public void RetrieveWebArtifact()
             Assert.AreEqual("http://test.fhir.org/r3/StructureDefinition/Observation", artifact.GetOrigin());
         }
 
-        private class TestFhirClient : Rest.FhirClient
+        private class TestFhirClient : LegacyFhirClient
```
Comment (marcovisserFurore) [inline src/Hl7.Fhir.Specification.Tests/Source/ResourceResolverTests.cs:92]: Why not using the new client for this?

File: src/Hl7.Fhir.Specification.Tests/Source/TerminologyTests.cs [hunk 2]
```diff
@@ -202,7 +203,7 @@ public void LocalTermServiceValidateCodeTest()
         [Fact(Skip = "Don't want to run these kind of integration tests anymore"), Trait("TestCategory", "IntegrationTest")]
         public void ExternalServiceValidateCodeTest()
         {
-            var client = new FhirClient("https://ontoserver.csiro.au/stu3-latest");
+            var client = new LegacyFhirClient("https://ontoserver.csiro.au/stu3-latest");
```
Comment (marcovisserFurore) [inline src/Hl7.Fhir.Specification.Tests/Source/TerminologyTests.cs:206]: Why not using the new client for this?

File: src/Hl7.Fhir.Specification/Specification/Source/WebResolver.cs [hunk 3]
```diff
@@ -24,25 +25,25 @@ public class WebResolver : IResourceResolver, IAsyncResourceResolver
         /// <summary>Default request timeout in milliseconds.</summary>
         public const int DefaultTimeOut = 5000;
 
-        readonly Func<Uri, FhirClient> _clientFactory;
+        readonly Func<Uri, LegacyFhirClient> _clientFactory;
```
Comment (marcovisserFurore) [inline src/Hl7.Fhir.Specification/Specification/Source/WebResolver.cs:28]: I would use here the new client.

**Comment by mmsmits** [comment]
fixes #1402

File: src/Hl7.Fhir.Specification.Tests/Source/ResourceResolverTests.cs [hunk 4]
```diff
@@ -89,48 +89,50 @@ public void RetrieveWebArtifact()
...
-            }
+            public TestFhirClient(string endpoint, FhirClientSettings settings = null, HttpMessageHandler handler = null) : base(endpoint, settings, handler) { Status = 1; }
         }
 
         [TestMethod, TestCategory("IntegrationTest")]
         public void RetrieveWebArtifactCustomFhirClient()
         {
-            TestFhirClient client = null;
+            using (var handler = new HttpClientEventHandler())
+            {
```
Comment (marcovisserFurore) [inline src/Hl7.Fhir.Specification.Tests/Source/ResourceResolverTests.cs:109]: Nested using can be without opening brace `{`. Like:
```
using (var handler = new HttpClientEventHandler())
using (var client = new TestFhirClient("http://test.fhir.org", handler: handler))
{
// etc.
}
```
or in C# 8 you can do this:
```
using var handler = new HttpClientEventHandler();
using var client = new TestFhirClient("http://test.fhir.org", handler: handler);
// etc.
```

File: src/Hl7.Fhir.Specification.Tests/Source/ResourceResolverTests.cs [hunk 5]
```diff
@@ -89,48 +89,50 @@ public void RetrieveWebArtifact()
...
+                    var artifact = wa.ResolveByUri("http://vonk.fire.ly/StructureDefinition/Patient");
 
-            Assert.IsNotNull(artifact);
-            Assert.IsTrue(artifact is StructureDefinition);
-            Assert.AreEqual("Patient", ((StructureDefinition)artifact).Name);
+                    Assert.IsNotNull(client);
+                    Assert.AreEqual(client.Status, 3);
+
+                    Assert.IsNotNull(artifact);
+                    Assert.IsTrue(artifact is StructureDefinition);
```
Comment (marcovisserFurore) [inline src/Hl7.Fhir.Specification.Tests/Source/ResourceResolverTests.cs:132]: If you use pattern matching, you don't have to cast the next line:
```
Assert.IsTrue(artifact is StructureDefinition sd);
Assert.AreEqual("Patient", sd.Name);
```

**Pull request merged**